Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ArchiveWSController.java

```diff
@@ -51,106 +53,32 @@ protected void initBinder(WebDataBinder binder)
 	}
 	
 	/**
-	 * This calls workspaceManger to list the workspace associated with a project for archival process.
-	 * @param   model
-	 * @param   projectid
-	 * @return  String - URL of the form
-	 * @throws  QuadrigaStorageException
-	 * @author  Kiran Kumar Batna
+	 This method archives the workspace for the project
+	 * 
+	 * @param  projectid                : Project identifier
+	 * @param  workspaceid              : Workspace identifier
+	 * @param  principal                : User for the workspace
+	 * @param  redirectAttributes       : Attributes to be forwarded to another controller
+	 * @return Model containing view
+	 * @throws QuadrigaStorageException
+	 * @throws QuadrigaAccessException
+	 * @author Kiran Kumar Batna
 	 */
-	@AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT,paramIndex = 1, userRole = {RoleNames.ROLE_COLLABORATOR_ADMIN,RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN,RoleNames.ROLE_PROJ_COLLABORATOR_CONTRIBUTOR})
-			,@ElementAccessPolicy(type=CheckedElementType.WORKSPACE,paramIndex=0,userRole={})})
-	@RequestMapping(value="auth/workbench/{projectid}/archiveworkspace", method=RequestMethod.GET)
-	public ModelAndView archiveWorkspaceForm(@PathVariable("projectid") String projectid,Principal principal) throws QuadrigaStorageException
-	{
-		ModelAndView model;
-		String userName;
-		ModifyWorkspaceForm workspaceForm;
-		List<ModifyWorkspace> archiveWorkspaceList;
-		
-		model = new ModelAndView("auth/workbench/workspace/archiveworkspace");
-		userName = principal.getName();
-		
-		workspaceForm = workspaceFormFactory.createModifyWorkspaceForm();
-		
-		// retrieve the workspaces associated with the projects
-		archiveWorkspaceList = workspaceFormManager.getActiveWorkspaceList(projectid, userName);
-		workspaceForm.setWorkspaceList(archiveWorkspaceList);
-		
-		model.getModelMap().put("workspaceform", workspaceForm);
-		model.getModelMap().put("wsprojectid",projectid);
-		model.getModelMap().put("success", 0);
-		return model;
-	}
+	@AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT,paramIndex = 1, 
+	userRole = {RoleNames.ROLE_COLLABORATOR_ADMIN, RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN, RoleNames.ROLE_PROJ_COLLABORATOR_CONTRIBUTOR}),
+	@ElementAccessPolicy(type=CheckedElementType.WORKSPACE,paramIndex=2,userRole={})})

```

Comment:
You probably want to add some user roles here. Only workspace admins should be allowed to do this.